### ISSUE TITLE
Offset global setting prevents footer elements from firing

### ISSUE THREAD

**Issue by Concept211** [matthieua/WOW#160]
When you set the global offset property at, let's say 200px and then you have animated elements near the bottom of the page that's under the 200px mark, it causes those elements to never animate.

**Comment by mdeboer** [comment]
Well that's because those elements need to be at least 200px in view to animate, as expected. If I'm correct you rather see all elements being animated when at the bottom?

I can imagine unless you set a "bottom margin" things would animate all of a sudden but only when you reach the ultimate bottom of the page. A "bottom margin" would probably be needed to be implemented to trigger all leftover the animations when reaching the bottom for things to go smoothly. Otherwise people scrolling to like 1px above the bottom of the page still wouldn't see the animations appearing.

Not something I have the time for to fix now but maybe I'll have a look at this later, if no one has picked this up in the meanwhile.

**Comment by Concept211** [comment]
Yes, that's correct - when you reach the very bottom of the window all left over elements should animate. 

The whole purpose of setting the offset is to show enough of the elements so that users can actually see and appreciate the animations, which is great. However, the downside is that those items at the very bottom of the page will never fire because the offset is never reached. 

I wouldn't want to add additional margins to the layout since I don't want to sacrifice the layout design simply to animate a couple of elements in the footer.

Thanks for replying and I would greatly appreciate a fix when you get a free moment.

Thanks again!

**Comment by mdeboer** [comment]
You're welcome!

With bottom margin I actually meant the bottom triggering offset. Say, "trigger all leftover animations when x pixels above the bottom of the page.". Because sometimes people don't scroll all the way down 0 px from bottom but instead end up say 10px above the bottom. It'll be optional though and default to 0, but I have to play with that.

**Comment by Concept211** [comment]
You're right - I totally see what you mean. I didn't think of that (people not fully hitting the bottom edge). Let me/us know whenever you have an updated.

Love the script, btw -- major time saver!

**Comment by kat2014** [comment]
hmmm. i am also having this problem and now am not sure how to resolve my issue. i thought setting 0px offset would do the trick, but as you say, some people do not reach so low on the page ever....
